### DISCUSSION TITLE
is there any limit or timeout for loading the csv

### DISCUSSION THREAD

**Discussion by agkcn** [grafana/grafana-csv-datasource#178] [Q&A]
Have a very large csv, ~100k rows. Unable to load using the datasource. Is there any limit  or timeout which can be tuned to load the entire csv ?

**Comment by voruti** [comment]
I think, I got the same problem.

Inside the Grafana "Query inspector" there is an error:
> error:"context deadline exceeded (Client.Timeout or context cancellation while reading body)"

I added https://covid.ourworldindata.org/data/owid-covid-data.csv as data source and want to get data from it ...

**Comment by marcusolsson** [comment]
Even if you increase the timeout, I'm not sure how well Grafana would perform with that amount of data being sent to the browser on each query.

We could introduce some form of server-side filtering to avoid sending the entire file on each query, but you'd still need to download it to the server (unless we cache it on the server).

For very large datasets, I'd probably consider ingesting them into a proper database.

**Comment by gdassori** [comment]
Hello, may I bump this? :-) 
I don't have a large dataset, but it is processed server-side and cached, so the first request may be slow and would often incur timeouts.

**Comment by kaenovsky** [comment]
Hi o/ anyone had news regarding this thread? In my case the csv is not THAT large but still it takes some time to load, I'd like to limit the number of rows. I'll follow this in case there's any update.
